.github/workflows/pre-commit.yml

```diff
@@ -2,9 +2,12 @@ name: Code formatting
 
 # see: https://help.github.com/en/actions/reference/events-that-trigger-workflows
 on:  # Trigger the workflow on push or pull request, but only for the main branch
-  push:
-    branches: [main]
-  pull_request: {}
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Why limit this to the main branch?

Reply:
Is there a scenario in which a pull request is directed towards a branch other than the main branch?

The change is based on this (https://github.com/microsoft/autogen/pull/946/files/e2fd5a014dfc656fa647c431d8d71ac917814264#diff-ac5eead3f3ce4863c524fff031a87b7aecccb4a0493df087a4e1c704a1505036L4).

Reply:
why remove the "push" trigger?

Reply:
"Is there a scenario in which a pull request is directed towards a branch other than the main branch?" I guess there could be such scenarios. Say I am working on an issue in branch "issue-a", and someone may want to make a PR to "issue-a" to contribute code with my review and approval.

Reply:
> why removing the "push" trigger?

I feel the original trigger was added by mistake. in `.pre-commit-config.yaml`, the direct push to main/master branch is disallowed by `no-commit-to-branch`. So this workflow will fail all the time when triggered by pushing to main, which is the only target branch to triggered by `push` event.

Also, this workflow will be triggered by `pull_request` event and will be ran against all PR towards `main branch`. From that aspect it's also surplus to run another format checking workflow after PR branch get merged

Reply:
> "Is there a scenario in which a pull request is directed towards a branch other than the main branch?" I guess there could be such scenarios. Say I am working on an issue in branch "issue-a", and someone may want to make a PR to "issue-a" to contribute code with my review and approval.

That's a good one, I can remove the branch filter if you want to support that scenraio.

Reply:
`merge_group` event is what you are looking for

The [`build`](https://github.com/microsoft/autogen/blob/a0288e081bf207baf93c12f3c2376a16f7521164/.github/workflows/build.yml#L16) workflow uses that trigger so it will be kicked off when adding to a merge queue

Reply:
@LittleLittleCloud  let's remove the branch filter as it's precomit?

Reply:
@ekzhu the branch filter is already removed in latest commit